mempool/src/shared_mempool.rs

```diff
@@ -259,6 +270,151 @@ async fn process_incoming_transactions<V>(
     notify_subscribers(SharedMempoolNotification::NewTransactions, &smp.subscribers);
 }
 
+/// send callback here
+async fn process_rpc_submit_transactions_request<V>(
+    smp: SharedMempool<V>,
+    peer_id: PeerId,
+    request: BroadcastTransactionsRequest,
+    callback: oneshot::Sender<Result<Bytes, RpcError>>,
+) where
+    V: TransactionValidation,
+{
+    // the RPC response object that will be returned to sender via callback
+    let resp = match submit_transactions_to_mempool(smp, peer_id, request).await {
+        Ok(response) => response,
+        Err(e) => {
+            error!(
+                "[shared mempool] Error occurred in submitting transactions to local mempool: {:?}",
+                e
+            );
+            // TODO return error in response
+            let mut response = BroadcastTransactionsResponse::default();
+            response.backpressure_ms = 0; // TODO this is placeholder
+            response
+        }
+    };
+    let response_msg = MempoolSyncMsg {
+        message: Some(MempoolSyncMsg_oneof::BroadcastTransactionsResponse(resp)),
+    };
+
+    // send response to callback
+    let response_data = response_msg.to_bytes().expect("failed to serialize proto");
+    if let Err(err) = callback
+        .send(Ok(response_data))
+        .map_err(|_| format_err!("[shared mempool] handling inbound RPC call timed out"))
+    {
+        error!(
+            "[shared mempool] failed to process batched transaction request, error: {:?}",
+            err
+        );
+    }
+}
+
+/// Tries to add transactions to local mempool
+/// returns RPC response that can be sent to callback
+/// peer_id is the PeerId of the node that sent the request
+async fn submit_transactions_to_mempool<V>(
```

Comment:
Mark process_incoming_transactions as having to get deprecated since these two are essentially the same thing with different responses